File: pmd/java/genericos/Caixa.java
package genericos;

import java.util.ArrayList;

public class Caixa<T> {

  private T conteudo;

  public Caixa(T conteudo) {
    this.conteudo = conteudo;
  }

  public T getConteudo() {
    return conteudo;
  }

  public void setConteudo(T conteudo) {
    this.conteudo = conteudo;
  }

  public static void main(String[] args) {
    Caixa<Chocolate> c2 = new Caixa<>(new Chocolate());
    Caixa<Geladeira> c3 = new Caixa<>(new Geladeira());
    ArrayList<Float> lista = new ArrayList<>();

  }

}

File: pmd/python/genericos/caixa.py
from typing import TypeVar, Generic

T = TypeVar('T')

class Caixa(Generic[T]):
  def __init__(self, conteudo:T): 
    self.conteudo = conteudo
  def getConteudo(self) -> T: 
    return self.conteudo
  def setConteudo(self, conteudo:T): 
    self.conteudo = conteudo

c1 = Caixa[str]("AGUA")
ct = c1.getConteudo()
print(ct)

c2 = Caixa[float](50.0)
ct = c2.getConteudo()
print(ct)
